source/docs/advanced/theme/route-dispatcher.md

```diff
@@ -22,103 +22,118 @@ If you come from a Magento background, this is the concept behind [URL Rewrites]
+  },
+};
 
-1. Create a custom GraphQL module that will add new `resolvers` to your GraphQL Schema See [Create a new GraphQL module](/docs/essentials/extend-the-graphql-schema.html) for more details.
-2. Create a loader that will match a URL string with an Entity (the object returned below):
-```js
-const MyModuleUrlLoader = makeDataLoader => () => {
-  return {
-    load: url => {
-      if (url === "my-dynamic-custom-url") {
-        return {
-          // The url that you are currently matching
-          url: "my-dynamic-custom-url",
-          // The type of entity you are willing to
-          // display for this URL
-          type: "myPageType",
-          // The id that will let you load the
-          // correct entity within your page component
-          identifier: "identifier",
-          // The path that will let you know what's
-          // the supposed URL before the URL rewrite
-          target_path: "catalog/product/view/id/17",
-          // A type that will let you know if its the
-          // correct URL or if you should redirect
-          // it. If it is a redirection, it should have
-          // the value `301`.
-          redirect_type: 0
-        }
-      }
-    }
-  }
-}
+export default moduleDefinition;
 ```
-    <blockquote class="note">
-    See [Loaders](/docs/advanced/graphql/slim-down-resolvers-with-loaders.html) documentation to learn how to instantiate your loader and add it to your GraphQL context.
-    </blockquote>
 
-    Here, we only handle `my-dynamic-custom-url` URL. However, you will most likely need to fetch the result from one of your backend instead. If you do so, the final result should still match the structure of an Entity as presented above.
+```diff
+// urlMatcher.js
++const myCustomUrlMatcher = (loader) => async (url, match) => {
++  if(match.__typename) {
++    // a match have been already loaded (you can still override it or return it as is like we do here)
++    return match;
++  }
++  const myMatch = await loader.loadByUrl(url);
++  if(!myMatch) {
++    return match; // return original match if none is found (this is needed!)
++  }
++  return { // return your custom routable entity
++    path: <canonical_url>,      // depending on your use case can be same as url
++    url,
++    __typename: 'MyCustomType', // the GraphQL type of your custom type (should be added in schema.gql),
++    ...myMatch, // or some adaptation of it. this will be available to your component defined bellow on a prop called `matched`
++  }
++};
++
++export default myCustomUrlMatcher;
+```
 
-3. Override the `route` query in order to use your brand new Loader.
-```js
-const resolvers = {
-  Query: {
-    route: (_, { url }, { loaders }) => {
-      return loaders.MyModuleUrl.matchBy(url)
-        .then((result) => {
-          if (result) {
-            return result;
-          } else {
-            return loaders.Url.matchBy(url);
-          }
-        })
-    },
-  }
+```diff
+# schema.gql
++MyCustomType {
```

Comment:
This type must implements the `Routable` interface and thus have à `path: String` field.

I think it should be part of the example.